tests/testHtmIndex.py

```diff
@@ -0,0 +1,213 @@
+        self.assertTrue(self.test_butler.datasetExists(dataset_name, data_id))
+        ref_cat = self.test_butler.get(dataset_name, data_id)
+        ex1 = ref_cat.extract('*')
+        ex2 = self.test_cat.extract('*')
+        # compare sets as the order may be different
+        self.assertEqual(set(ex1.keys()), set(ex2.keys()))
+        for key in ex1:
+            self.assertTrue(numpy.array_equal(ex1[key], ex2[key]))
+
+    def testIngest(self):
+        default_config = IngestIndexedReferenceTask.ConfigClass()
+        #test ingest with default config
+        #This should raise since I haven't specified the ra/dec/mag columns.
+        self.assertRaises(ValueError, IngestIndexedReferenceTask.parseAndRun, args=[input_dir, "--output",
+                          self.out_path+"/output", self.sky_catalog_file], config=default_config)
+        #test with ~minimum config.  Mag errors are not technically necessary, but might as well test here
+        default_config.ra_name = 'ra_icrs'
+        default_config.dec_name = 'dec_icrs'
+        default_config.mag_column_list = ['a', 'b']
+        default_config.mag_err_column_map = {'a':'a_err'}
+        #should raise since all columns need an error column if any do
+        self.assertRaises(ValueError, IngestIndexedReferenceTask.parseAndRun, args=[input_dir, "--output",
+                          self.out_path+"/output", self.sky_catalog_file], config=default_config)
+        #test with multiple files and correct config
+        default_config.mag_err_column_map = {'a':'a_err', 'b':'b_err'}
+        retVal = IngestIndexedReferenceTask.parseAndRun(args=[input_dir, "--output", self.out_path+"/output_multifile",
+                                      self.sky_catalog_file, self.sky_catalog_file], 
+                                      config=default_config)
+        #test with config overrides
+        default_config = IngestIndexedReferenceTask.ConfigClass()
+        default_config.ra_name = 'ra'
+        default_config.dec_name = 'dec'
+        default_config.mag_column_list = ['a', 'b']
+        default_config.mag_err_column_map = {'a':'a_err', 'b':'b_err'}
+        default_config.ref_dataset_name = 'other_photo_astro_ref'
+        default_config.level = 10
+        default_config.is_photometric_name = 'is_phot'
+        default_config.is_resolved_name = 'is_res'
+        default_config.is_variable_name = 'is_var'
+        default_config.id_name = 'id'
+        default_config.extra_col_names = ['val1', 'val2', 'val3']
+        default_config.header_lines = 1
+        default_config.colnames = ['id', 'ra', 'dec', 'a', 'a_err', 'b', 'b_err', 'is_phot', 'is_res',
+                                   'is_var', 'val1', 'val2', 'val3']
+        default_config.delimiter = '|'
+        #this also tests changing the delimiter
+        retVal = IngestIndexedReferenceTask.parseAndRun(args=[input_dir, "--output", self.out_path+"/output_override",
```

Comment:
This test is testing that things don't crash? Is there any way to test that the right thing happened?